code/modules/mob/living/simple_animal/hostile/statue.dm

```diff
@@ -65,6 +65,8 @@
 	mob_spell_list += new /obj/effect/proc_holder/spell/aoe_turf/flicker_lights(src)
 	mob_spell_list += new /obj/effect/proc_holder/spell/aoe_turf/blindness(src)
 	mob_spell_list += new /obj/effect/proc_holder/spell/targeted/night_vision(src)
+	teleport = new
```

Comment:
the action's target should be src, if it is it will autocleanup when src is deleted